File: _bench/fib/fib.zig
const std = @import("std");

pub fn main() void {
    const n: u64 = 40;
    std.debug.print("{d}\n", .{fib(n)});
}

fn fib(n: u64) u64 {
    if (n < 2) {
        return 1;
    } else {
        return fib(n - 1) + fib(n - 2);
    }
}

File: _bench/fib/fib.ex
defmodule Fib do
  def fib(0), do: 1
  def fib(1), do: 1

  def fib(n) when n >= 2 do
    fib(n - 1) + fib(n - 2)
  end

  def main() do
    n = 40
    :io.format("~p~n", [fib(n)])
  end
end
